### PR TITLE
Context multithreaded test

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**

### CODE REVIEW

File: test/graft_server_test.cpp [hunk 1]
```diff
@@ -293,98 +300,98 @@ class GraftServerTest : public ::testing::Test
...
+                EXPECT_EQ(s, in);
+            }
+            if(bl)
+            {
+                std::string s = ctx.local[in];
+                EXPECT_EQ(s, in);
+            }
+        };
+
+        auto pre = [&](const graft::Router::vars_t& vars, const graft::Input& input, graft::Context& ctx, graft::Output& output)->graft::Router::Status
```
[the remark this answers is not in the document]
Comment (AlexanderSuprunenko) [reply test/graft_server_test.cpp]: As for me, they are self-explanatory already. We have triple Handlers. The peri is called by a worker in threadPool, pre and post, if they are not empty, are called by the io-thread before and after it places a job to threadPool. 
I can name them as preHandler... or preFunction as you decide.

[the remark this answers is not in the document]
Comment (mbg033) [reply test/graft_server_test.cpp]: what does "peri" means? Yes, as for me I would prefer to name it with nouns.

[the remark this answers is not in the document]
Comment (AlexanderSuprunenko) [reply test/graft_server_test.cpp]: Have fixed. thanks

File: include/router.h [hunk 2]
```diff
@@ -42,19 +42,19 @@ class RouterT
         Handler3& operator = (Handler3&&) = default;
         ~Handler3() = default;
 
-        Handler3(const Handler& pre, const Handler& peri, const Handler& post)
-            : pre(pre), peri(peri), post(post)
+        Handler3(const Handler& pre_action, const Handler& action, const Handler& post_action)
+            : pre_action(pre_action), action(action), post_action(post_action)
```
Comment (EDDragonWolf) [inline include/router.h]: `action(action)`
Maybe, better to rename it to `thread_action` or `worker_action` or something like this.

**Pull request merged**